### PR TITLE
Rebuild search indices in chunks

### PR DESCRIPTION
**Pull request by kaedroho**
Wagtail currently reindexes each model in one go. If a model is very large, this can quick fill up the RAM. There is also no indication of progress

This pull request makes Wagtail index 1000 objects at a time and output a '.' in the console for each 1000 objects indexed.

### CODE REVIEW

File: wagtail/wagtailsearch/management/commands/update_index.py [hunk 1]
```diff
@@ -69,3 +84,47 @@ def handle(self, **options):
...
+
+    def print_newline(self):
+        self.stdout.write('')
+
+    def print_iter_progress(self, iterable):
+        """
+        Print a progress meter while iterating over an iterable. Use it as part
+        of a ``for`` loop::
+
+            for item in self.progress(big_long_list):
```
Comment (gasman) [inline wagtail/wagtailsearch/management/commands/update_index.py]: `progress` -> `print_iter_progress`

File: wagtail/wagtailsearch/management/commands/update_index.py [hunk 2]
```diff
@@ -69,3 +84,47 @@ def handle(self, **options):
...
+
+            self.stdout.flush()
+
+        self.print_newline()
+
+    # Atomic so the count of models doesnt change as it is iterated
+    @transaction.atomic
+    def queryset_chunks(self, qs, chunk_size=1000):
+        """
+        Yield a queryset in chunks of at most ``chunk_size``. The chunk yeilded
```
Comment (gasman) [inline wagtail/wagtailsearch/management/commands/update_index.py]: yeilded -> yielded

**Comment by gasman** [comment]
Besides those comment fixes (and the things being addressed in #1490), looks good to me!

**Comment by kaedroho** [comment]
Thanks!

Typos fixed

**Pull request merged**